doc/tui_rstgen.py

```diff
@@ -0,0 +1,306 @@
+        Fluent session mode either ``meshing`` or ``solver``.
+
+    Returns
+    -------
+        TUI file name.
+    """
+    for file in os.listdir(_get_tui_file_path(mode)):
+        if fnmatch.fnmatch(file, "tui_*.py"):
+            return pathlib.Path(file).stem
+
+
+def _get_sorted_members(members):
+    """Sort members alphabetically.
+
+    Parameters
+    ----------
+    members: list
+        Attributes of ``TUIMenu`` or ``TUIMethod``.
+    """
+    return sorted([member.__name__ for member in members])
+
+
+def _write_doc(menu, mode):
+    """Write .rst file for each menu.
+
+    Parameters
+    ----------
+    menu: type
+        TUIMenu, TUIMethod
+    mode: str
+        Fluent session mode either ``meshing`` or ``solver``.
+    """
+    menu_name, menu_path = _get_menu_name_path(menu["name"])
+    full_folder_path = _get_tui_docdir(mode, menu_path)
+    Path(full_folder_path).mkdir(parents=True, exist_ok=True)
+    index_file = Path(full_folder_path) / "index.rst"
+    with open(index_file, "w", encoding="utf8") as f:
+        f.write(f".. _ref_{mode}_tui_{menu['name'].__name__}:\n\n")
+        f.write(f"{menu['name'].__name__}\n")
+        f.write(f"{'=' * len(menu['name'].__name__)}\n\n")
+        f.write(f".. autoclass:: {menu_name}\n")
+        f.write(
+            f"   :members: {', '.join(_get_sorted_members(menu['without_members']))}\n"
+        )
+        f.write("   :show-inheritance:\n")
+        f.write("   :undoc-members:\n")
+        f.write("   :autosummary:\n")
+        f.write("   :autosummary-members:\n\n")
+
+        if menu["with_members"]:
+            f.write(".. toctree::\n")
+            f.write("   :hidden:\n\n")
+            for member in _get_sorted_members(menu["with_members"]):
+                f.write(f"   {member}/index\n")
+
+
+def _generate_all_attribute_classes(Menus, main_menu):
+    """Set all attribute classes with and without members as attributes to ``Menus``
+    class.
+
+    Parameters
+    ----------
+    Menus: type
+        ``Menus`` class.
+    main_menu: type
+        ``main_menu`` class.
+    """
+    with_members, without_members = _get_attribute_classes_with_and_without_members(
+        main_menu
+    )
+    setattr(
+        Menus,
+        main_menu.__name__,
+        dict(
+            name=main_menu, with_members=with_members, without_members=without_members
+        ),
+    )
+    for member in with_members:
+        _get_attribute_classes_recursively(member, Menus)
+
+
+def _generate_doc(Menus, mode):
+    """Write .rst file for each attribute class.
+
+    Parameters
+    ----------
+    Menus: type
+        ``Menus`` class.
+    mode: str
+        Fluent session mode either ``meshing`` or ``solver``.
+    """
+    all_menus_attrs = [attr for attr in dir(Menus) if not attr.startswith("__")]
+    for attr in all_menus_attrs:
+        menu_attr = getattr(Menus, attr)
+        _write_doc(menu_attr, mode)
+
+
+def generate(main_menu, mode):
+    """Generate .rst files.
```

Comment:
```suggestion
    """Generate result files.
```

Reply:
Done. Thank you.

Reply:
.rst are not results file. `.rst` is fine or we can say - `Generate RST files.`

Reply:
Done. Thank you.